tests/test_events.py

```diff
@@ -327,3 +332,46 @@ def test_event_is_upcoming(event, mocker):
 
     with freeze_time(tomorrow_morning):
         assert not test_event.is_upcoming
+
+
+def test_delete_button_shows(event, client, django_user_model, mocker):
+    # create 1 event with a fake api_source and use requests_mock to come up with 2 different responses (an ok and a 404)
+    e = event.build()
+    event_template = reverse('lametro:events', args=[e.slug])
+    api_source = f'http://webapi.legistar.com/v1/metro/events/{e.slug}'
+    
+    mock_source = mocker.MagicMock()
+    mock_source.url = api_source
+    mock_api_source = mocker.patch('lametro.models.LAMetroEvent.api_source', new_callable=mocker.PropertyMock, return_value=mock_source)
+
+    username = 'admin'
+    password = 'administrativepw'
+    user = django_user_model.objects.create_user(username=username, password=password)
+    client.force_login(user)
```

Comment:
Good thought! `pytest-django` provides an `admin_client` fixture that will do this setup for you: https://pytest-django.readthedocs.io/en/latest/helpers.html#admin-client-django-test-client-logged-in-as-admin

What do you think about switching?

Reply:
definitely a lot more concise to use that fixture!